tests/helpers/testers.py

```diff
@@ -116,16 +116,23 @@ def _class_test(
     metric = pickle.loads(pickled_metric)
 
     for i in range(rank, NUM_BATCHES, worldsize):
-        batch_result = metric(preds[i], target[i])
+        b_kwargs_update = {k: v[i] if isinstance(v, Tensor) else v for k, v in kwargs_update.items()}
```

Comment:
is there a reason for rename the new var as `b_...`? maybe just a better name, not sure what b is for...

Reply:
stands for batch, just thought adding "batch" would make the variable name a little long (can definitely change it if needed)

Reply:
ok, so lets keep it as `batch_kwargs_update `

Reply:
```suggestion
        batch_kwargs_update = {k: v[i] if isinstance(v, Tensor) else v for k, v in kwargs_update.items()}
```